Use <hostname>.toml if local.toml is not found

NomisIV opened the pull request:
Not fully tested nor optimized, but it seems to work.
Let me know if you want me to restructure anything (like the location of the `file_exists()` function)

Ideally we would only check the hostname if `local.toml` isn't found, but I ran into lifetime errors since the hostname variable would expire before loading the config, unless I duplicated the code for loading the file (one if local.toml was found, and another if it wasn't).

SuperCuber commented:
I'd write this slightly differently:
```rust
// have local_config be mutable
if !local_config.exists() {
    info!("{:?} not found, using hostname.toml instead", local_config);
    let hostname = ...;
    local_config.set_file_name(&format!("{}.toml", hostname));
}
```
My slight concern with this is that the error message that happens in case they are both missing may be confusing to someone who is using a `local.toml` and isn't expecting dotter to also look in that location.

NomisIV commented:
Your method looks a lot better, and I'm almost ashamed of what I came up with lol.
I have modified it slightly, because I needed to convert local_config into a pathbuf to be able to modify it. And I also changed "hostname.toml" to actually display your hostname, and also explain that it is your hostname.
Tested and it seems to work still

NomisIV commented:
Fixed the formatting now. Is there anything else keeping this PR from merging?

SuperCuber commented:
Looks good to me! Thanks for the contribution :D
